source/acquaman/AMScanConfiguration.cpp

```diff
@@ -137,3 +141,28 @@ void AMScanConfiguration::setAxisControlInfos(const AMControlInfoList &axisContr
 		emit axisControlInfosChanged();
 	}
 }
+
+void AMScanConfiguration::setAutomaticDirectionAssessment(bool isAutomatic)
+{
+	if (automaticDirectionAssessment_ != isAutomatic){
+
+		automaticDirectionAssessment_ = isAutomatic;
+		emit automaticDirectionAssessmentChanged(automaticDirectionAssessment_);
+		setModified(true);
+	}
+}
+
+void AMScanConfiguration::setDirection(AMScanConfiguration::Direction newDirection)
+{
+	if (direction_ != newDirection){
+
+		direction_ = newDirection;
+		emit directionChanged(direction_);
+		setModified(true);
+	}
+}
+
+void AMScanConfiguration::setDirection(int newDirection)
+{
+	setDirection(Direction(newDirection));
```

Comment:
That is a good point.  But the cast still just casts it to an int.  I don't think enum casts modify values except in standard int(double) casts.